File: Abstractions/Types/TeamModel.cs
﻿namespace Timeoff.Types
{
    public record TeamModel
    {
        public string Name { get; init; } = null!;
        public double Allowance { get; init; }
        public bool IncludePublicHolidays { get; init; }

        public bool IsAccruedAllowance { get; init; }

        public int? ManagerId { get; init; }

        public ResultModels.ListResult? Manager { get; init; }
    }
}

File: timeoff.ui/src/app/services/teams/team.model.ts
import { UserModel } from '../company/user.model';

export interface TeamModel {
    id: number;
    name: string;
    manager: UserModel;
    allowance: number;
    employeeCount: number;
    includePublicHolidays: boolean;
    isAccruedAllowance: boolean;
}
